utils/wasm-gen/src/generator/syscalls/imports.rs

```diff
@@ -242,36 +242,91 @@
 }
 
 impl<'a, 'b> SysCallsImportsGenerator<'a, 'b> {
-    /// Generates a function which calls "properly" the `gr_reservation_send`.
-    fn generate_send_from_reservation(&mut self) {
-        let (reserve_gas_idx, reservation_send_idx) = {
-            let maybe_reserve_gas = self
-                .sys_calls_imports
-                .get(&InvocableSysCall::Loose(SysCallName::ReserveGas))
-                .map(|&(_, call_indexes_handle)| call_indexes_handle);
-            let maybe_reservation_send = self
-                .sys_calls_imports
-                .get(&InvocableSysCall::Loose(SysCallName::ReservationSend))
-                .map(|&(_, call_indexes_handle)| call_indexes_handle);
-
-            match maybe_reserve_gas.zip(maybe_reservation_send) {
-                Some(indexes) => indexes,
+    /// Returns the indexes of invocable sys-calls.
+    fn invocable_sys_calls_indexes<const N: usize>(
+        &self,
+        sys_calls: [SysCallName; N],
+    ) -> Option<[usize; N]> {
+        let mut indexes = [0; N];
+        let iter = sys_calls.iter().map(|&sys_call| {
+            self.sys_calls_imports
+                .get(&InvocableSysCall::Loose(sys_call))
+                .map(|&(_, call_indexes_handle)| call_indexes_handle)
+        });
+
+        for (index, maybe_index) in indexes.iter_mut().zip(iter) {
+            match maybe_index {
+                Some(idx) => *index = idx,
                 None => {
-                    log::trace!("Wasm hasn't got either gr_reserve_gas, or gr_reservation_send, or both of them");
-                    return;
+                    log::trace!(
+                        "The following sys-calls must be imported: {missing_sys_calls:?}",
+                        missing_sys_calls = sys_calls.map(|sys_call| sys_call.to_str()),
+                    );
+                    return None;
                 }
             }
-        };
+        }
+
+        Some(indexes)
+    }
```

Comment:
**Suggestion:** Consider refactoring the `invocable_sys_calls_indexes` method to return a `Result` instead of `Option`. This would allow you to provide more detailed error information when a sys-call is not found, which could be helpful for debugging.
```suggestion
    fn invocable_sys_calls_indexes<const N: usize>(&self, sys_calls: [SysCallName; N]) -> Result<[usize; N], String> {
        let mut indexes = [0; N];
        let iter = sys_calls.iter().map(|&sys_call| {
            self.sys_calls_imports.get(&InvocableSysCall::Loose(sys_call)).map(|&(_, call_indexes_handle)| call_indexes_handle)
        });

        for (index, maybe_index) in indexes.iter_mut().zip(iter) {
            match maybe_index {
                Some(idx) => *index = idx,
                None => {
                    let missing_sys_calls = sys_calls.map(|sys_call| sys_call.to_str());
                    return Err(format!("The following sys-calls must be imported: {:?}", missing_sys_calls));
                }
            }
        }

        Ok(indexes)
    }
```

Reply:
cool, but it is easier to use `return;` instead of logging string/missing_syscall every time